src/main.rs

```diff
@@ -1,12 +1,43 @@
 use bevy::prelude::*;
+use crate::Direction::Up;
 
 fn main() {
     App::new()
         .add_plugins(DefaultPlugins)
         .add_startup_system(setup)
+        .add_system(list_snakes)
         .run();
 }
 
+#[derive(Component)]
+struct Snake;
+
+type XCoord = u8;
+type YCoord = u8;
+
+#[derive(Component, Default, Debug)]
+struct Location {
+    x: XCoord,
+    y: YCoord,
+}
+
+#[derive(Default, Debug)]
+enum Direction {
+    #[default]
+    Up, Down, Left, Right,
+}
+
+#[derive(Component, Default, Debug)]
+struct Moving {
+    dir: Direction,
+}
+
+fn list_snakes(query: Query<(&Location, &Moving), With<Snake>>) {
+    for (loc, moving) in query.iter() {
+        println!("Snake: {:?}, {:?}", loc, moving);
+    }
+}
+
```

Comment:
Maybe this shouldn't be in main, but in different module(s)?

Reply:
Have moved stuff into sub-modules, and I know there is a way to encapsulate this more by making it a plugin.  Happy for someone else to add their ideas.

Reply:
Looks way better in my opinion.

Reply:
Cool - want to add input or movement each tick?

Reply:
Sure, I can try to